config.yaml

```diff
@@ -2,39 +2,67 @@
 # See LICENSE file for licensing details.
 #
 options:
-  ssh-user:
-    default: "ubuntu"
-    description: Specify an SSH user for sos collect to connect to nodes with.
+  model-name:
+    type: string
+    default: ""
+    description: |
+      The model where the apps/units/machines resides. This will be the default
+      to use when "model-name" is not specified in "collect" action.
+  upload-method:
+    type: string
+    default: "sftp"
+    description: |
+      The method to use when uploading sosreports to the remote server.
+      Supported methods are "[sftp, scp, http]" (Default to "sftp").
+  juju-username:
     type: string
-  units:
     default: ""
-    description: Provide a comma-delimited list of units to collect sosreports from.
+    description: ""
+  juju-password:
     type: string
-  model:
     default: ""
+    description: ""
+  juju-endpoint:
     type: string
+    default: ""
+    description: ""
   server:
-    #default: "https://files.support.canonical.com/uploads/"
+    type: string
     default: "files.support.canonical.com"
-    description: Specify a server to upload the sosreport.
-    type: string 
-  server-username:
-    default: "ubuntu"
-    description: ""
-    type: string 
-  server-password:
+    description: |
+      Specify the server to upload the sosreport.
+  username:
+    type: string
     default: "ubuntu"
-    description: ""
+    description: |
+      Specify the username for server.
+  password:
```

Comment:
Should we group upload parameters by type?

Reply:
I have not check if all upload parameters are the same for all type `scp` / `sftp` / `http`, this is currently specific for `sftp`. I think at the first stage, we should not complicate the MR by generalizing the parameters since we haven't implement them yet.